beetmoverscript/script.py

```diff
@@ -154,6 +156,43 @@
     copy_beets(context, candidates_keys_checksums, releases_keys_checksums)
 
 
+async def push_to_maven(context):
+    """Push artifacts to locations expected by maven clients (like mvn or gradle)"""
+    context.artifacts_to_beetmove = task.get_upstream_artifacts_with_zip_extract_param(context)
+    context.release_props = get_release_props(context)
+    context.checksums = dict()  # Needed by downstream calls
+    context.raw_balrog_manifest = dict()    # Needed by downstream calls
+
+    mapping_manifest = generate_beetmover_manifest(context)
+    validate_bucket_paths(context.bucket, mapping_manifest['s3_bucket_path'])
+
+    expected_files = maven.get_maven_expected_files_per_archive_per_task_id(
+        context.artifacts_to_beetmove, mapping_manifest
+    )
+
+    extracted_paths_per_archive = zip.check_and_extract_zip_archives(
+        context.artifacts_to_beetmove,
+        expected_files,
+        context.config.get('zip_extract_max_file_size_in_mb', ZIP_MAX_FILE_SIZE_IN_MB)
+    )
+
+    number_of_extracted_archives = len(extracted_paths_per_archive)
+    if number_of_extracted_archives == 0:
+        raise ScriptWorkerTaskException('No archive extracted')
+    elif number_of_extracted_archives > 1:
+        raise NotImplementedError('More than 1 archive extracted. Only 1 is supported at once')
+    extracted_paths_per_relative_path = list(extracted_paths_per_archive.values())[0]
+
+    context.artifacts_to_beetmove = {
```

Comment:
I was wondering if it's worth moving this `context.artifacts_to_beetmove` logic + logic from 169-179 in a separate function. Again, no strong feelings, we can leave them as they are, I'm more leaning towards consistency with the other `push_to_*` operations.

Reply:
Yeah, I was wondering the same. I can't find a name for this function. I get the feeling too many concepts are intertwined. That's why I left it in this function. How would you describe this function?

Reply:
Hm, good question. I wonder, how bad it would be if we defined that function within push_to_maven()? Something like:

```
async def push_to_maven(context):
       def _process_and_validate_artifacts(...)
        ....
        return artifacts
        
        ......
       context.artifacts_to_beetmove = _process_and_validate_artifacts(...)
```

I know most of the other similar functions are cleaner but if we don't find any better solution, this could be a good compromise. Alternatively, we can leave things be, I don't want to block on this, nor I have strong feelings.

Reply:
Done in https://github.com/mozilla-releng/beetmoverscript/pull/163/commits/7f5682ee62a71f78ea11e1e8ac4b4550d713f409. Unit tests are failing because of #176.

Tested e2e in https://tools.taskcluster.net/groups/FN8cz6cTSyiKeVxa8Y0KAg

Reply:
Looks great, thanks for making this amendment!